### PR TITLE
Rollup Updates and Improvements

### PR DESCRIPTION
**Pull request by vladtar**
* Updated most `Rollup` packages
* Consolidated the way we do `date-fns`, `FontAwesome`, `lodash` imports
* Code split `prettier` into its own bundle and dynamically importing it now - it would be nice to be able to do langchain like that as well, in the next iteration
* For the `AIAssistant` now only using `highlight.js` with `js` instead of all languages
* Misc cleanup
* Fixed a bunch of typescript errors, some of which probably need review

Bundle sizes now:

### Note about imports

Rollup properly tree-shakes `date-fns` and `FontAwesome`, so:

YES ✔️  👇 
```
import { faExclamationTriangle, faCheckCircle, faRefresh } from "@fortawesome/free-solid-svg-icons";
```

NO 🚫  👇 (more verbose, though roughly same bundle size)
```
import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons/faExclamationTriangle";
import { faCheckCircle } from "@fortawesome/free-solid-svg-icons/faCheckCircle"
import { faRefresh } from "@fortawesome/free-solid-svg-icons/faRefresh"
```

YES ✔️  👇 
```
import { formatRelative, formatDistanceToNow, parseISO, add } from "date-fns";
```

NO 🚫  👇 
```
import formatRelative from 'date-fns/formatRelative'
import formatDistanceToNow from 'date-fns/formatDistanceToNow'
import parseISO from 'date-fns/parseISO'
import add from 'date-fns/add'
```

Lodash does NOT properly tree-shake, at least not the base version that we are using:

YES ✔️  👇 
```
import ceil from 'lodash/ceil'
```

NO 🚫  👇 
```
import { ceil } from 'lodash'
```

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [ ] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

## Summary by CodeRabbit

- **New Features**
  - Enhanced build process with the addition of new plugins for improved JavaScript minification and TypeScript declaration generation.

- **Bug Fixes**
  - Updated several dependencies to their latest versions, which may include important bug fixes and performance improvements.

- **Refactor**
  - Streamlined import statements across various components for better organization and readability.
  - Simplified the `NavigationBar` component's interface by removing `onBack` and `onClose` props.

- **Chores**
  - Removed unnecessary TypeScript compiler options related to declaration file generation in the project configuration.

### CODE REVIEW

File: src/components/container/AIAssistant/AIAssistant.tsx
Comment (vladtar) [inline src/components/container/AIAssistant/AIAssistant.tsx:165]: Dynamic imports

File: src/components/presentational/ResourceList/ResourceList.tsx
Comment (vladtar) [inline src/components/presentational/ResourceList/ResourceList.tsx:38]: @greinard 👆 ?

File: src/components/view/ExternalAccountsView/ExternalAccountsView.tsx
Comment (vladtar) [inline src/components/view/ExternalAccountsView/ExternalAccountsView.tsx:59]: @chrisnowak 👆 ?

File: tsconfig.json
Comment (vladtar) [inline tsconfig.json:17]: Doing this in the separate:

```
{
		input: "src/index.ts",
		output: [{ file: "dist/index.d.ts", format: "esm" }],
		plugins: [dts()],
		external: [/\.css$/, /\.stories.tsx$/, /\.previewData.ts$/]
	}
```

otherwise Rollup and Typescript and the fact that there were 2 outputs which are now directories really confused each other.

File: package.json
Comment (vladtar) [inline package.json:33]: Removed this because it's in `peerDependencies`.

File: package.json
Comment (vladtar) [inline package.json:46]: Moved this from regular `dependencies` to `devDependencies`.

File: package.json
Comment (greinard) [reply package.json:33]: I was always wondering if I was missing some reason why it was duplicated here.  😄

File: src/components/container/SurveyAnswerChart/SurveyAnswerData.previewdata.tsx
Comment (greinard) [reply src/components/container/SurveyAnswerChart/SurveyAnswerData.previewdata.tsx:2]: I think `Duration` is being used down on line 26.

File: src/components/presentational/Chat/Chat.tsx
Comment (greinard) [inline src/components/presentational/Chat/Chat.tsx:36]: Minor:  Formatting seems a tad off here.

File: src/components/presentational/ResourceList/ResourceList.tsx
Comment (greinard) [reply src/components/presentational/ResourceList/ResourceList.tsx:38]: Yeah, I must have missed removing this line at some point.  The absence of a button variant means to hide the button now, so it was probably just leftover during that refactor.

Removing this is correct.  👍

File: src/components/view/ExternalAccountsView/ExternalAccountsView.tsx
Comment (greinard) [reply src/components/view/ExternalAccountsView/ExternalAccountsView.tsx:59]: Looks like those functions (`onBack` and `onClose`) can probably be removed from the properties above too.

There is only one usage of `onBack`, and I don't think it is needed anymore.

File: src/components/container/ConnectedDevices/ConnectedDevices.tsx
Comment (greinard) [inline src/components/container/ConnectedDevices/ConnectedDevices.tsx:6]: I think you need to drop the `/` at the end of this import or the storybook fails to load.

**Review by greinard — CHANGES_REQUESTED** [review]
Looks pretty good to me.  I went through the storybook some to try out a few different components.  I didn't specifically test the AI components, which appear to have undergone the biggest adjustment, dependency-wise, but if you are confident they are good to go, I am good with that.

The only change I think we need to make before merging is the removal of that trailing slash on the font-awesome import I pointed out.  I'll keep an eye out to retry the storybook once you've pushed that change.

File: src/components/container/ConnectedDevices/ConnectedDevices.tsx
Comment (vladtar) [reply src/components/container/ConnectedDevices/ConnectedDevices.tsx:6]: Nice find. Removed.

File: src/components/view/ExternalAccountsView/ExternalAccountsView.tsx
Comment (vladtar) [reply src/components/view/ExternalAccountsView/ExternalAccountsView.tsx:59]: Looks like this is still being used in https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/view/StandaloneHealthAndWellnessView/StandaloneHealthAndWellnessView.tsx#L189.

Comment (greinard) [reply src/components/view/ExternalAccountsView/ExternalAccountsView.tsx:59]: Yes, but the `onBack` handler on the `ExternalAccountsViewProps` that is being used there is never actually being called by the `ExternalAccountsView`, so I think the "back" functionality is working because of this:  https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/view/StandaloneHealthAndWellnessView/StandaloneHealthAndWellnessView.tsx#L75

```
 MyDataHelps.back = back;
```

So I think we can remove those handler functions from the props, since they aren't being called, and just clean up the code above that (the code you linked) so that it no longer passes a handler that gets ignored.

**Review by greinard — APPROVED** [review]
Storybook is working for me now.  The `ExternalAccountsViewProps` cleanup is probably quick, but not urgent, so I'll leave that up to you whether to do it in this PR or not.

Feel free to merge when ready.  :ship:

**Pull request merged**
